src/actions/write/evolveState.ts

```diff
@@ -25,24 +16,14 @@ export const evolveState = async (
     throw new ContractError(NON_CONTRACT_OWNER_MESSAGE);
   }
 
-  const { epochStartHeight, epochEndHeight } = getEpochDataForHeight({
-    currentBlockHeight: new BlockHeight(+SmartWeave.block.height),
-    epochZeroStartHeight: new BlockHeight(
-      state.distributions.epochZeroStartHeight,
-    ),
-    epochBlockLength: new BlockHeight(EPOCH_BLOCK_LENGTH),
-  });
-
-  const prescribedObservers = await getPrescribedObserversForEpoch({
-    gateways: state.gateways,
-    distributions: state.distributions,
-    epochStartHeight,
-    epochEndHeight,
-    minOperatorStake: GATEWAY_REGISTRY_SETTINGS.minOperatorStake,
-  });
-
-  state.prescribedObservers = {
-    [epochStartHeight.valueOf()]: prescribedObservers,
```

Comment:
Why does all this disappear when adding the bark record?

Reply:
we execute `evolveState` to add new attributes to state. `prescribedObservers` has already been added and is now modified/updated via `tick` interactions. it's similar to a DB migration where it only needs to be run once.